### PR TITLE
Add private ctor for singleton classes.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/correlationcontext/DefaultCorrelationContextManagerProvider.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.correlationcontext;
+
+import io.opentelemetry.correlationcontext.spi.CorrelationContextManagerProvider;
+
+public final class DefaultCorrelationContextManagerProvider
```
Comment (jkwatson) [inline api/src/main/java/io/opentelemetry/correlationcontext/DefaultCorrelationContextManagerProvider.java]: looks like this snuck into this PR. Can it be moved to another PR, since it doesn't seem to be directly related to it?

Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/correlationcontext/DefaultCorrelationContextManagerProvider.java]: Lazy to update this now, will get updated once https://github.com/open-telemetry/opentelemetry-java/pull/960 is merged, so I will wait to merge that one first.

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/correlationcontext/DefaultCorrelationContextManagerProvider.java]: aha, cool.  👍

Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/correlationcontext/DefaultCorrelationContextManagerProvider.java]: done :)
